### PR TITLE
[Explicit Module Builds] Avoid adding cross-target input dependencies to intermediate build artifact jobs.

### PR DESCRIPTION
**Pull request by artemcm**
For a given target, in the Explicit Module Build world, `swift-driver` produces jobs to build the main module and jobs to build all module dependencies.
Cross-target dependencies should be added as explicit inputs to a target's main module build job, but not intermediate build dependency jobs.

Always adding cross-target dependencies as inputs can lead to cycles in the build manifest when multiple targets share common intermediate module dependencies.

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci smoke test

File: Sources/Build/ManifestBuilder.swift [hunk 1]
```diff
@@ -218,22 +219,35 @@ extension LLBuildManifestBuilder {
...
 
+                // Add target dependencies as inputs to the main module build command.
+                //
+                // Jobs for a target's intermediate build artifacts, such as PCMs or
+                // modules built from a .swiftinterface, do not have a
+                // dependency on cross-target build products. If multiple targets share
+                // common intermediate dependency modules, such dependencies can lead
+                // to cycles in the resulting manifest.
+                var manifestNodeInputs = jobInputs
+                if !job.inputs.contains(where: { $0.type == .swiftInterface }) &&
```
Comment (DougGregor) [inline Sources/Build/ManifestBuilder.swift:230]: Should this predicate be given a name and sunk into swift-driver itself?

Comment (artemcm) [reply Sources/Build/ManifestBuilder.swift:230]: Yeah, that's a good idea.

**Comment by artemcm** [comment]
@swift-ci smoke test

**Pull request merged**